### ISSUE TITLE
Custom cluster fails to come up with RHEL 7.6 nodes having upstream docker 18.09 

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#19299]
Rancher version : v2.0.13

K8s version: v1.11.9
Docker version : 18.09
OS : RHEL 7.6

Steps
1. Create a 3 node custom cluster with RHEL 7.6 nodes and upstream docker 18.09. Choose k8s version v1.11.9
N1 : All roles
N2: All roles
N3: All roles

One node gets registered. Other two nodes are stuck in registering state. Cluster shows as active.

Note: Cluster with RHEL 7.5 and upstream docker 18.09 with K8s version v1.11.9 came up successfully

Logs below:
```
2019/04/01 18:55:22 [INFO] cluster [c-z8vdq] provisioning: [healthcheck] service [kube-proxy] on host [abcd] is healthy
2019/04/01 18:55:24 [INFO] cluster [c-z8vdq] provisioning: [worker] Successfully started [rke-log-linker] container on host [abcd]
2019/04/01 18:55:24 [INFO] cluster [c-z8vdq] provisioning: [remove/rke-log-linker] Successfully removed container on host [abcd]
2019/04/01 18:55:24 [INFO] cluster [c-z8vdq] provisioning: [worker] Successfully started Worker Plane..
2019/04/01 18:55:24 [INFO] cluster [c-z8vdq] provisioning: [sync] Syncing nodes Labels and Taints
2019/04/01 18:55:24 [INFO] cluster [c-z8vdq] provisioning: [sync] Successfully synced nodes Labels and Taints
2019/04/01 18:55:24 [INFO] cluster [c-z8vdq] provisioning: [network] Setting up network plugin: canal
2019/04/01 18:55:24 [INFO] cluster [c-z8vdq] provisioning: [addons] Saving addon ConfigMap to Kubernetes
2019/04/01 18:55:25 [INFO] cluster [c-z8vdq] provisioning: [addons] Successfully Saved addon to Kubernetes ConfigMap: rke-network-plugin
2019/04/01 18:55:25 [INFO] cluster [c-z8vdq] provisioning: [addons] Executing deploy job..
2019/04/01 18:55:30 [INFO] cluster [c-z8vdq] provisioning: [addons] Setting up KubeDNS
2019/04/01 18:55:30 [INFO] cluster [c-z8vdq] provisioning: [addons] Saving addon ConfigMap to Kubernetes
2019/04/01 18:55:31 [INFO] cluster [c-z8vdq] provisioning: [addons] Successfully Saved addon to Kubernetes ConfigMap: rke-kubedns-addon
2019/04/01 18:55:31 [INFO] cluster [c-z8vdq] provisioning: [addons] Executing deploy job..
2019/04/01 18:55:36 [INFO] cluster [c-z8vdq] provisioning: [addons] KubeDNS deployed successfully..
2019/04/01 18:55:36 [INFO] cluster [c-z8vdq] provisioning: [addons] Setting up Metrics Server
2019/04/01 18:55:36 [INFO] cluster [c-z8vdq] provisioning: [addons] Saving addon ConfigMap to Kubernetes
2019/04/01 18:55:36 [INFO] cluster [c-z8vdq] provisioning: [addons] Successfully Saved addon to Kubernetes ConfigMap: rke-metrics-addon
2019/04/01 18:55:36 [INFO] cluster [c-z8vdq] provisioning: [addons] Executing deploy job..
2019/04/01 18:55:47 [INFO] cluster [c-z8vdq] provisioning: [addons] KubeDNS deployed successfully..
2019/04/01 18:55:47 [INFO] cluster [c-z8vdq] provisioning: [ingress] Setting up nginx ingress controller
2019/04/01 18:55:47 [INFO] cluster [c-z8vdq] provisioning: [addons] Saving addon ConfigMap to Kubernetes
2019/04/01 18:55:47 [INFO] cluster [c-z8vdq] provisioning: [addons] Successfully Saved addon to Kubernetes ConfigMap: rke-ingress-controller
2019/04/01 18:55:47 [INFO] cluster [c-z8vdq] provisioning: [addons] Executing deploy job..
2019/04/01 18:55:58 [INFO] cluster [c-z8vdq] provisioning: [ingress] ingress controller nginx is successfully deployed
2019/04/01 18:55:58 [INFO] cluster [c-z8vdq] provisioning: [addons] Setting up user addons
2019/04/01 18:55:58 [INFO] cluster [c-z8vdq] provisioning: [addons] no user addons defined
2019/04/01 18:55:58 [INFO] cluster [c-z8vdq] provisioning: Finished building Kubernetes cluster successfully
2019/04/01 18:56:01 [INFO] Provisioned cluster [c-z8vdq]
2019/04/01 18:56:01 [ERROR] ClusterController c-z8vdq [cluster-agent-controller] failed with : ClusterUnavailable 503: cluster not found
2019/04/01 18:56:01 [INFO] Provisioning cluster [c-z8vdq]
2019/04/01 18:56:01 [INFO] Creating cluster [c-z8vdq]
2019/04/01 18:56:01 [INFO] Cluster c-z8vdq already exists.
2019/04/01 18:56:01 [INFO] Provisioned cluster [c-z8vdq]
2019/04/01 18:56:01 [INFO] Starting cluster agent forc-z8vdq
2019/04/01 18:56:01 [INFO] Creating user for principal system://c-z8vdq
2019/04/01 18:56:01 [INFO] registering podsecuritypolicy cluster handler for cluster c-z8vdq
2019/04/01 18:56:01 [INFO] registering podsecuritypolicy project handler for cluster c-z8vdq
2019/04/01 18:56:01 [INFO] registering podsecuritypolicy namespace handler for cluster c-z8vdq
2019/04/01 18:56:01 [INFO] registering podsecuritypolicy serviceaccount handler for cluster c-z8vdq
2019/04/01 18:56:01 [INFO] registering podsecuritypolicy template handler for cluster c-z8vdq
2019/04/01 18:56:01 [INFO] Starting cluster controllers for c-z8vdq
2019/04/01 18:56:01 [INFO] [mgmt-auth-crtb-controller] Creating clusterRoleBinding for membership in cluster c-z8vdq for subject u-6xzthzg4ih
2019/04/01 18:56:04 [ERROR] ClusterAlertController c-z8vdq/clusteralert-controllermanager [cluster-alert-deployer] failed with : Delete https://abcd:6443/api/v1/namespaces/cattle-alerting?timeout=30s: dial tcp 127.0.0.1:6443: getsockopt: connection refused
2019/04/01 18:56:04 [ERROR] ClusterAlertController c-z8vdq/clusteralert-deploment-event [cluster-alert-deployer] failed with : Delete https://abcd:6443/api/v1/namespaces/cattle-alerting?timeout=30s: dial tcp 127.0.0.1:6443: getsockopt: connection refused
2019/04/01 18:56:04 [ERROR] ClusterAlertController c-z8vdq/clusteralert-etcd [cluster-alert-deployer] failed with : Delete https://abcd:6443/api/v1/namespaces/cattle-alerting?timeout=30s: dial tcp 127.0.0.1:6443: getsockopt: connection refused
2019/04/01 18:56:04 [ERROR] ClusterRoleTemplateBindingController c-z8vdq/u-6xzthzg4ih-admin [cluster-crtb-sync] failed with : couldn't ensure roles: couldn't create role cluster-owner: clusterroles.rbac.authorization.k8s.io "cluster-owner" already exists
2019/04/01 18:56:17 [INFO] Handling backend connection request [c-z8vdq]
2019/04/01 18:56:17 [INFO] Updating pod [ingress-nginx/nginx-ingress-controller-8nc7v] with public endpoints [[{"nodeName":"c-z8vdq:m-eec239e40a54","addresses":["abcd"],"port":80,"protocol":"TCP","podName":"ingress-nginx:nginx-ingress-controller-8nc7v","allNodes":false},{"nodeName":"c-z8vdq:m-eec239e40a54","addresses":["abcd"],"port":443,"protocol":"TCP","podName":"ingress-nginx:nginx-ingress-controller-8nc7v","allNodes":false}]]
2019/04/01 18:56:18 [INFO] Updating workload [ingress-nginx/nginx-ingress-controller] with public endpoints [[{"nodeName":"c-z8vdq:m-eec239e40a54","addresses":["abcd"],"port":80,"protocol":"TCP","podName":"ingress-nginx:nginx-ingress-controller-8nc7v","allNodes":false},{"nodeName":"c-z8vdq:m-eec239e40a54","addresses":["abcd"],"port":443,"protocol":"TCP","podName":"ingress-nginx:nginx-ingress-controller-8nc7v","allNodes":false}]]


```
From the logs, it was observed that cluster was deployed only on one of the nodes

**Comment by alena1108** [comment]
@soumyalj can you confirm that the workaround described in https://github.com/rancher/rancher/issues/19305#issuecomment-479695114, works for this issue too.

**Comment by soumyalj** [comment]
Follow up Observation for the above issue:

When RHEL 7.6 nodes with upstream docker are added one at a time by allowing each node to come to active state(before adding the next one), the custom cluster gets provisioned correctly.

**Comment by alena1108** [comment]
This issue has the same root cause as the issues documented in RN for 2.1.x release line:

https://github.com/rancher/rancher/issues/15970
https://github.com/rancher/rancher/issues/15969
https://github.com/rancher/rancher/issues/15695

When custom nodes get registered in parallel, there are chances of cluster being stuck in updating/provisioning state. It is not RHEL 7.6 specific, and happens for RHEL 7.5 as well. The issue is fixed in 2.2.0 by refactoring etcd and state management for RKE clusters.
